### ISSUE TITLE
AGPS contribution

### ISSUE THREAD

**Issue by ernesst** [ubports/ubports-touch#92]
Enhancement (low priority)

Allow an UBphone using MLS to contribute to the service when GPS is used if user agree or could be an app development. 
In a area with low coverage, a user could contribute for his future use.

**Comment by NeoTheThird** [comment]
I agree, this would be a nice feature. This is the android equivalent: https://blog.mozilla.org/services/2014/11/03/mozilla-stumbler-1-0/

**Comment by ernesst** [comment]
It's also implemented in background in firefox for android

**Comment by ernesst** [comment]
Is there a chance to get GPS information in background ? We refurbished Activity tracker app which create gpx files. Currently it cannot get gps info if screen off... 
Otherwise this software could a base where we need to merge cell + wifi + gps info into a file to upload to Mozilla.

**Comment by userj** [comment]
> Is there a chance to get GPS information in background ? We refurbished Activity tracker app which create gpx files. Currently it cannot get gps info if screen off...
> Otherwise this software could a base where we need to merge cell + wifi + gps info into a file to upload to Mozilla.

Ideally, I would suggest that one would be able to put the phone in airplane mode, but still turn on the GPS and get the information in the background.

**Comment by ernesst** [comment]
I've made a wrapper to collect the data and contribute : 
https://forums.ubports.com/topic/118/mozilla-locations-service-on-ubuntu-touch/7
https://framagit.org/ernesst/MLS-for-Utouch

It's capturing GPS + WIFI and upload the info to MLS server at the end. 

I couldn't find a way to get Cell tower information, ofono script doesn't allow that on the phone. If you have any idea please let me know.

**Comment by dobey** [comment]
> I couldn't find a way to get Cell tower information, ofono script doesn't allow that on the phone. If you have any idea please let me know.

There's some code in location-service already, to report data to mozilla location service. It would need some integration work and such to finish u, but we could probably use it instead.

**Comment by ernesst** [comment]
@dobey 
If you could give an hint where this piece of code is located it might help future developments.

**Comment by dobey** [comment]
@ernesst It's part of location-service code, though it's currently unused in production as it's incomplete. It would need appropriate settings integration, so that people can opt-in to the location reporting. It's C++ and a bit esoteric.
